test/lager_test_function_transform.erl

```diff
@@ -0,0 +1,152 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2011-2017 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+-module(lager_test_function_transform).
+
+-include("lager.hrl").
+
+-lager_function_transforms([
+  {returns_static, {lager_test_function_transform, transform_static}},
+  {returns_dynamic, {lager_test_function_transform, transform_dynamic}},
+  {returns_undefined, {not_real_module_fake, fake_not_real_function}}
+]).
+
+-compile({parse_transform, lager_transform}).
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-export([
+  transform_static/0,
+  transform_dynamic/0
+]).
+-endif.
+
+-ifdef(TEST).
+
+transform_static() ->
+  static_result.
+
+transform_dynamic() ->
+  erlang:monotonic_time().
```

Comment:
This is going to make all tests that aren't running OTP 19 fail as Travis shows.  One solution would be to use `lager_util:otp_version/0` in your tests to make the appropriate call to the function that is correct depending on the OTP version.